lib/dry/types/coercions.rb

```diff
@@ -102,6 +102,36 @@ def to_symbol(input, &block)
       def empty_str?(value)
         EMPTY_STRING.eql?(value)
       end
+
+      ISO_DATE = /\A(\d{4})-(\d\d)-(\d\d)\z/.freeze
+      def fast_string_to_date(string)
+        if string =~ ISO_DATE
+          ::Date.new $1.to_i, $2.to_i, $3.to_i
```

Comment:
This is not thread-safe so it cannot use these nasty regexp globals

Reply:
this is actually is. These are pseudo-global

Reply:
https://ruby-doc.org/core-2.7.2/Regexp.html#class-Regexp-label-Special+global+variables

_These global variables are thread-local and method-local variables._